autoscaler/controllers/secret_controller.go

```diff
@@ -62,10 +35,12 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	return ctrl.Result{}, nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
 func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Secret{}).
-		WithEventFilter(&secretPredicate{}).
+		// we need to handle secrets only when they are updated.
+		// this is to trigger redeployment of the cluster collector in case of destination secret change.
+		// when the secret was just created (via auto-scaler restart or initial deployment), the cluster collector will be reconciled by other controllers.
+		WithEventFilter(predicate.And(&odigospredicate.OnlyUpdatesPredicate{}, &predicate.ResourceVersionChangedPredicate{})).
```

Comment:
Should we use a generation change predicate instead of a resource version predicate here?

Reply:
I think secrets has no spec, thus no generation